lib/puppet_x/bodeco/util.rb

```diff
@@ -4,7 +4,7 @@ module PuppetX
   module Bodeco
     module Util
       def self.download(url, filepath, options = {})
-        uri = URI(url)
+        uri = Puppet::Util.path_to_uri(url) unless url.is_a?(URI)
```

Comment:
The unless will always be false since we removed the URI(url) line above. In this case if the Puppet::Util.path_to_uri is smart maybe just remove the unless. Need to double check their code to be sure.